tests/testthat/test-tags.r

```diff
@@ -677,6 +677,14 @@ test_that("Latin1 and system encoding are converted to UTF-8", {
 
   expect_identical(Encoding(format(HTML(latin1_str))), "UTF-8")
   expect_identical(Encoding(format(tagList(latin1_str))), "UTF-8")
+
+  # ensure the latin1 attribute returns correctly after escaping
+  latin1_str2 <- rawToChar(as.raw(c(0xff, 0x0d, 0x0a)))
+  Encoding(latin1_str2) <- "latin1"
+  expect_identical(
```

Comment:
Can you change these tests to use the same pattern as above? That is, use `charToRaw()` and then compare the raw output to a specific byte sequence. Also, test that the encoding is UTF-8.

The reason it's important to do tests this way is because Unicode escaped values like `\u00ff` can be encoded in different ways, so simply using a string like `"\u00ff"` does not specify a byte sequence. Also, `expect_identical()` uses `identical()`, which automatically does encoding conversions. For example, on my Mac, I can run this:


```R
x <- rawToChar(as.raw(c(0xc3, 0xbf)))
Encoding(x) <- "UTF-8"

x
#> [1] "ÿ"
Encoding(x)
#> [1] "UTF-8"
charToRaw(x)
#> [1] c3 bf

x1 <- iconv(x, to="latin1")
x1
#> [1] "ÿ"
Encoding(x1)
#> [1] "latin1"
charToRaw(x1)
#> [1] ff

# This passes (!)
expect_identical(x, x1)
```

Reply:
Thanks for sharing the reason! I've changed the test.